include/RAJA/internal/MemUtils_CPU.hpp

```diff
@@ -102,12 +111,12 @@ inline T* allocate_aligned_type(size_t alignment, size_t size)
 ///
 inline void free_aligned(void* ptr)
 {
-#if defined(HAVE_POSIX_MEMALIGN)
+#if defined(RAJA_HAVE_POSIX_MEMALIGN) || defined(RAJA_HAVE_ALIGNED_ALLOC)
   free(ptr);
 #elif defined(RAJA_PLATFORM_WINDOWS)
   _aligned_free(ptr);
 #else
-#error "No known aligned allocator available"
+  free(((void**)ptr)[-1]);
```

Comment:
What does this do?

Reply:
segfaults :)

Reply:
`(*)(int*)(0) = 42`

is the canonical way of doing that

Reply:
If this is supposed to error, we should just throw an exception 😎

Reply:
This is a hazy recollection: do exceptions make linking iffier? I have a memory I don't trust that says they do

Reply:
The `aligned_allocate` stores the address of the base of the allocated region one slot behind the beginning of the data given to the user.  Basically it does this:

```c
// in alloc
void ** mem = malloc(...);
void ** usermem = mem + some_offset;
usermem[-1] = mem;
// in free
free(usermem[-1]);
```

Reply:
Thanks for the explanation. Can you add that as a comment in the code?

Reply:
@DavidPoliakoff not so much anymore, unless you're linking with C and you let one escape, then you're in for a bumpy ride...

Reply:
@trws 

Good to know, and thanks for the actual explanation